proposals/csharp-13.0/ref-struct-interfaces.md

```diff
@@ -233,33 +235,25 @@ Examples of these rules in action:
 ```csharp
 interface I1 { }
 I1 M1<T>(T p)
-    where T : allows ref struct, I1
+    where T : I1, allows ref struct
 {
     // Error: cannot box potential ref struct
     return p;
 }
 
-T M2<T>(T p)
+ref T M2<T>(ref T p)
     where T : allows ref struct
 {
-    Span<int> span = stackalloc int[42];
 
     // The safe-to-escape of the return is current method because one of the inputs is
     // current method
-    T t = M3<T>(span);
+    T t = M3<T>(default);
 
     // Error: the safe-to-escape is current method.
-    return t;
+    return ref t;
 
     // Okay
-    return default;
-    return p;
-}
-
-T M3<T>(Span<T> span)
-    where T : allows ref struct
-{
-    return default;
+    return ref p;
 }
```

Comment:
This is the sample that should get the most thorough review. Much of it didn't compile as written because `T` can't be allowed to be a `ref struct` if it will be used as a `Span<T>`. That meant much of `M2` needed edits.